### PR TITLE
linux-pam: Drop musl patch that’s applied upstream

### LINKED ISSUE #108692: pkgsMusl.linux-pam fails to build: patch musl-fix-pam_exec.patch out of date
With current master +  #108685 (fixing coreutils), the next issue I am facing is this:
```
~/build/nixpkgs $ nix-build -A pkgsMusl.linux-pam
these derivations will be built:
  /nix/store/0j569z807p9fjf0083xhzhjpps4vwc9h-linux-pam-1.5.1.drv
building '/nix/store/0j569z807p9fjf0083xhzhjpps4vwc9h-linux-pam-1.5.1.drv'...
unpacking sources
unpacking source archive /nix/store/cbbam7s59sqrsqk2dh25izkjmdngwd5h-Linux-PAM-1.5.1.tar.xz
source root is Linux-PAM-1.5.1
setting SOURCE_DATE_EPOCH to timestamp 1606323454 of file Linux-PAM-1.5.1/doc/specs/parse_y.h
patching sources
applying patch /nix/store/mbcmh0dbsczh53p7h5mcwbmynx1fqq1m-fix-compat.patch?id=05a62bda8ec255d7049a2bd4cf0fdc4b32bdb2cc
patching file modules/pam_lastlog/pam_lastlog.c
Hunk #2 succeeded at 52 with fuzz 1 (offset 3 lines).
applying patch /nix/store/nynvcd4ljflkxw6gb81dqdwwvidixzvg-libpam-fix-build-with-eglibc-2.16.patch?id=05a62bda8ec255d7049a2bd4cf0fdc4b32bdb2cc
patching file modules/pam_unix/pam_unix_acct.c
Hunk #1 succeeded at 50 (offset 3 lines).
applying patch /nix/store/r1f2scxc864wdy3r9mm1lx7gsy36ml0c-musl-fix-pam_exec.patch
patching file modules/pam_exec/pam_exec.c
Hunk #1 FAILED at 103.
Hunk #2 FAILED at 180.
2 out of 2 hunks FAILED -- saving rejects to file modules/pam_exec/pam_exec.c.rej
builder for '/nix/store/0j569z807p9fjf0083xhzhjpps4vwc9h-linux-pam-1.5.1.drv' failed with exit code 1
error: build of '/nix/store/0j569z807p9fjf0083xhzhjpps4vwc9h-linux-pam-1.5.1.drv' failed
```

This patch was added in “c260a8d0635 libpam: fix w/musl after update to 1.3.1” by @dtzWill.

Possibly broken by “d703c1ef25775dbe94e9120fe988db24c2df48c3 pam: 1.3.1 -> 1.5.1” by @NeQuissimus.

### PR DESCRIPTION
**Pull request by nomeata**
fixes: #108692

### CODE REVIEW

**Comment by nomeata** [comment]
Should be merged after https://github.com/NixOS/nixpkgs/pull/108685

**Comment by nomeata** [comment]
@tobim, since you look at the previous musl related fix, maybe you can check this one too?

File: pkgs/tools/misc/coreutils/default.nix [hunk 1]
```diff
@@ -76,8 +76,7 @@ stdenv.mkDerivation (rec {
   outputs = [ "out" "info" ];
 
   nativeBuildInputs = [ perl xz.bin ]
-    ++ optionals stdenv.hostPlatform.isCygwin [ autoreconfHook texinfo ]   # due to patch
-    ++ optionals stdenv.hostPlatform.isMusl [ autoreconfHook bison ];   # due to patch
+    ++ optionals stdenv.hostPlatform.isCygwin [ autoreconfHook texinfo ];  # due to patch
```
Comment (tobim) [inline pkgs/tools/misc/coreutils/default.nix:79]: This is already done in your other PR. Added by mistake?

**Review by tobim — CHANGES_REQUESTED** [review]
I didn't test this, but the upstream change indicates that this patch should indeed not be necessary anymore.

File: pkgs/tools/misc/coreutils/default.nix (hunk 1 above)
Comment (nomeata) [reply pkgs/tools/misc/coreutils/default.nix:79]: No, I plan to merge this after the other has gone in (I believe that this depends on the other to even get far enough)

Comment (tobim) [reply pkgs/tools/misc/coreutils/default.nix:79]: I didn't see that it is in its own commit. It should work then.

**Comment by alyssais** [comment]
Other patch is in now.

**Comment by nomeata** [comment]
> Other patch is in now.

Thanks, rebased

**Review by tobim — APPROVED** [review]
LGTM

**Review by knedlsepp — APPROVED** [review]
LGTM

**Comment by pjjw** [comment]
bleh sorry looks like #109906 got merged in front of this

**Comment by nomeata** [comment]
Nothing to be sorry about, as long as it gets fixed :-)

**Pull request closed without merging**